org.mwc.cmap.TimeController/src/org/mwc/cmap/TimeController/recorders/CoordinateRecorder.java

```diff
@@ -144,14 +149,297 @@ protected void showMessageDialog(final String message)
+
+    public CoordinateRecorder getRecorder()
+    {
+      final File testFile = new File(
+          "../org.mwc.cmap.combined.feature/root_installs/sample_data/offset_times.rep");
+
+      // ok, now try to read it in
+      final Layers _theLayers = new Layers();
+
+      // add the REP importer
+      MWC.Utilities.ReaderWriter.ImportManager.addImporter(
+          new Debrief.ReaderWriter.Replay.ImportReplay());
+
+      // get our thread to import this
+      final MWC.Utilities.ReaderWriter.ImportManager.BaseImportCaller reader =
+          new MWC.Utilities.ReaderWriter.ImportManager.BaseImportCaller(
+              new File[]
+              {testFile}, _theLayers)
+          {
+            // handle completion of the full import process
+            @Override
+            public void allFilesFinished(final File[] fNames,
+                final Layers newData)
+            {
+              Application.logError2(Application.INFO,
+                  "All files loaded correctly", null);
+            }
+
+            // handle the completion of each file
+            @Override
+            public void fileFinished(final File fName, final Layers newData)
+            {
+              Application.logError2(Application.INFO, "File " + fName
+                  + " loaded correctly", null);
+            }
+          };
+
+      // and start it running
+      reader.start();
+
+      // wait for the results
+      while (reader.isAlive())
+      {
+        try
+        {
+          Thread.sleep(100);
+        }
+        catch (final java.lang.InterruptedException e)
+        {
+          Application.logError2(Application.INFO, "Error while sleeping", e);
+        }
+      }
+
+      /*
+       * final PlainProjection projection = new org.mwc.cmap.gt2plot.proj.GtProjection();
+       */
+      final PlainProjection projection =
+          new MWC.Algorithms.Projections.FlatProjection();
+      projection.setScreenArea(new java.awt.Dimension(1443, 901));
+      projection.setDataArea(new WorldArea(new WorldLocation(22.238965795584505,
+          -21.928244631862952, 0), new WorldLocation(22.238965795584505,
+              -21.43985414608609, 0)));
+
+      final org.mwc.cmap.core.DataTypes.Temporal.TimeControlProperties timePreferences =
+          new org.mwc.cmap.core.DataTypes.Temporal.TimeControlProperties();
+
+      CoordinateRecorder recorder = new CoordinateRecorder(_theLayers,
+          projection, timePreferences);
+
+      return recorder;
+    }
+
+    /**
+     * COLLINGWOOD starts after NELSON
+     */
+    public void testPrimaryStartsFirst()
+    {
+      CoordinateRecorder recorder = getRecorder();
+
+      MWC.GenericData.HiResDate currentTime = new MWC.GenericData.HiResDate(
+          818748540000L);
+
+      final int AMOUNT_OF_STEPS = 3;
+      doIteration(recorder, currentTime, AMOUNT_OF_STEPS);
+
+      final java.util.Map<String, Debrief.ReaderWriter.powerPoint.model.Track> track =

```

Comment:
@saulhidalgoaular - I'm thinking out loud here.  I would prefer you to retrieve the first segment, then to perform a series of:
`assertEquals("correct course", 12.0f, segment.getCourse(), 0,0001`

This has the advantage that it will find the specific attribute that doesn't match, and it also removes the dependency on unrelated code : track-point generation, and the TrackPoint.equals() implementation.

Reply:
@saulhidalgoaular - sorry this is dragging on.  There's a subtle improvement to the way you've structured your tests.

If you use:
````
assertEquals("correct number of skipped steps", 1, track.get(collingwood),getStepsToSkip());
````
Then, when the test fails we get an explanation of what went wrong, plus it says "expected to receive `1`, but received `5`".  This makes it slightly quicker to diagnose what went wrong.

Reply:
oh Sir. I am sorry. I thought you meant about the attributes of the Track. I ignored it because you said below to remove it. I am doing now the changes you mentioned,